v2/api/dbformysql/v1/user_types.go

```diff
@@ -150,6 +152,104 @@ func (user *User) deleteValidations() []func() error {
 
 // updateValidations validates the update of the resource
 func (user *User) updateValidations() []func(old runtime.Object) error {
+	return []func(old runtime.Object) error{
+		func(old runtime.Object) error {
+			return user.validateIsLocalOrAAD()
+		},
+		user.validateUserTypeNotChanged,
+		user.validateWriteOnePropertiesNotChanged,
+		user.validateUserAADAliasNotChanged,
+	}
+}
+
+// validateWriteOnePropertiesNotChanged function validates the update on WriteOnce properties.
+// TODO: Note this should be kept in sync with admissions.ValidateWriteOnceProperties
+func (user *User) validateWriteOnePropertiesNotChanged(oldObj runtime.Object) error {
+	var errs []error
+
+	oldUser, ok := oldObj.(*User)
+	if !ok {
+		// This shouldn't happen, but if it does don't block things
+		return nil
+	}
+
+	// If we don't have a finalizer yet, it's OK to change things
+	hasFinalizer := controllerutil.ContainsFinalizer(oldUser, genruntime.ReconcilerFinalizer)
+	if !hasFinalizer {
+		return nil
+	}
+
+	if oldUser.Spec.AzureName != user.Spec.AzureName {
+		errs = append(errs, errors.Errorf("updating 'AzureName' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldUser.GetName()))
```

Comment:
Would it be clearer to separate creation of the error from the append? 
And maybe to use one parameter per line to increase visibility of the parameters used?

```suggestion
		err := errors.Errorf(
			"updating 'AzureName' is not allowed for '%s : %s", 
			oldObj.GetObjectKind().GroupVersionKind(), 
			oldUser.GetName())
		errs = append(errs, err)
```

Reply:
Done